GCDWebServer/Core/GCDWebServerConnection.m

```diff
@@ -548,6 +548,8 @@ - (void)_readRequestHeaders {
           }
         }
         if (_request) {
+          _request.localAddressData = [self.localAddressData copy];
+          _request.remoteAddressData = [self.remoteAddressData copy];
```

Comment:
Copy is not needed and it were it would have to be done by property setter not the caller / user of the property.